src/store/baking/baking-epics.ts

```diff
@@ -1,46 +1,46 @@
+import { Action } from '@reduxjs/toolkit';
+import retry from 'async-retry';
 import { combineEpics, Epic } from 'redux-observable';
-import { from, Observable, of } from 'rxjs';
+import { from, of } from 'rxjs';
 import { catchError, map, switchMap } from 'rxjs/operators';
-import { Action } from 'ts-action';
 import { ofType } from 'ts-action-operators';
 
 import { getTzktApi } from 'src/api.service';
-import { BakerInterface, bakingBadApi, fetchBaker, emptyBaker, buildUnknownBaker } from 'src/apis/baking-bad';
+import { BakerInterface, bakingBadApi, fetchBaker, buildUnknownBaker } from 'src/apis/baking-bad';
 import { BakerRewardInterface } from 'src/interfaces/baker-reward.interface';
 import type { RootState } from 'src/store/types';
-import { isTruthy } from 'src/utils/is-truthy';
 import { createReadOnlyTezosToolkit } from 'src/utils/rpc/tezos-toolkit.utils';
+import { RPC_RETRY_OPTIONS } from 'src/utils/tezos.util';
 import { withSelectedAccount, withSelectedRpcUrl } from 'src/utils/wallet.utils';
 
 import { loadBakerRewardsListActions, loadBakersListActions, loadSelectedBakerActions } from './baking-actions';
 
 const NUMBER_OF_BAKER_REWARDS_TO_LOAD = 30;
 
-const loadSelectedBakerAddressEpic: Epic = (action$: Observable<Action>, state$: Observable<RootState>) =>
+const loadSelectedBakerAddressEpic: Epic<Action, Action, RootState> = (action$, state$) =>
   action$.pipe(
     ofType(loadSelectedBakerActions.submit),
     withSelectedAccount(state$),
     withSelectedRpcUrl(state$),
-    switchMap(([[, selectedAccount], rpcUrl]) =>
-      from(createReadOnlyTezosToolkit(rpcUrl, selectedAccount).rpc.getDelegate(selectedAccount.publicKeyHash)).pipe(
-        switchMap(bakerAddress => {
-          if (!isTruthy(bakerAddress)) {
-            return of(emptyBaker);
-          }
+    switchMap(([[, selectedAccount], rpcUrl]) => {
+      const tezos = createReadOnlyTezosToolkit(rpcUrl, selectedAccount);
 
-          return from(fetchBaker(bakerAddress)).pipe(map(baker => baker || buildUnknownBaker(bakerAddress)));
-        }),
-        map(baker => loadSelectedBakerActions.success(baker)),
-        catchError(error => {
-          console.error(error);
+      return from(retry(() => tezos.rpc.getDelegate(selectedAccount.publicKeyHash), RPC_RETRY_OPTIONS));
+    }),
+    switchMap(bakerAddress =>
```

Comment:
Please consider getting baker's address from TZKT for the first attempt

Reply:
+